src/mixins/itext_behavior.mixin.ts

```diff
@@ -173,90 +174,77 @@ export abstract class ITextBehaviorMixin<
   /**
    * @private
    */
-  _tick() {
-    this._currentTickState = this._animateCursor(
-      this,
-      1,
-      this.cursorDuration,
-      '_onTickComplete'
-    );
-  }
-
-  /**
-   * @private
-   */
-  _animateCursor(obj, targetOpacity, duration, completeMethod) {
-    const tickState = {
-      isAborted: false,
-      abort: function () {
-        this.isAborted = true;
-      },
-    };
-
-    obj.animate('_currentCursorOpacity', targetOpacity, {
-      duration: duration,
-      onComplete: function () {
-        if (!tickState.isAborted) {
-          obj[completeMethod]();
-        }
-      },
-      onChange: function () {
-        // we do not want to animate a selection, only cursor
-        if (obj.canvas && obj.selectionStart === obj.selectionEnd) {
-          obj.renderCursorOrSelection();
-        }
+  _animateCursor({
+    toValue,
+    duration,
+    delay,
+    onComplete,
+  }: {
+    toValue: number;
+    duration: number;
+    delay?: number;
+    onComplete?: TOnAnimationChangeCallback<number, void>;
+  }) {
+    return animate({
+      startValue: this._currentCursorOpacity,
+      endValue: toValue,
+      duration,
+      delay,
+      onComplete,
+      abort: () => {
+        return (
+          !this.canvas ||
+          // we do not want to animate a selection, only cursor
+          this.selectionStart !== this.selectionEnd
+        );
       },
-      abort: function () {
-        return tickState.isAborted;
+      onChange: (value) => {
+        this._currentCursorOpacity = value;
+        this.renderCursorOrSelection();
       },
     });
```

Comment:
Yes animatableObject is a convenience wrapper, but in this case here animate is good enough because you are inside a class, you can use a fat arrow on the onChange to change this properties and you are animating a single thing.
```js
rect.animate({ angle, left, top }, { duration:  200 })
```
Is really comfortable.

Is not that we need to have it at all costs, but is not comparable with `canvas.exitEditing vs canvas.textManager.exitEditing`;